test/e2e/framework.go

```diff
@@ -1209,6 +1216,29 @@ func (data *TestData) getSparkOperator() (string, error) {
 	return pod.Name, nil
 }
 
+// getGrafanaPod retrieves the name of the Grafana Pod (grafana-*).
+func (data *TestData) getGrafanaPod() (string, error) {
```

Comment:
Function `getGrafanaPod()` and `getSparkOperator()` duplicate most of code, consider to build a general `getPodName()` function instead?